src/common-utils/main.sh

```diff
@@ -437,6 +437,10 @@ if [ "${INSTALL_ZSH}" = "true" ]; then
     fi
 
     if [ "${CONFIGURE_ZSH_AS_DEFAULT_SHELL}" == "true" ]; then
+        # Fixing chsh always asking for a password on alpine linux
+        # ref: https://askubuntu.com/questions/812420/chsh-always-asking-a-password-and-get-pam-authentication-failure.
+        echo "auth sufficient pam_shells.so" > /etc/pam.d/chsh
```

Comment:
Also, if the user already has `auth required pam_shells.so` in `/etc/pam.d/chsh`, then shouldn't that be replaced with `auth sufficient pam_shells` for `chsh` to work?

Reply:
Yes, that's correct. If that's the case then we can change it back after `chsh` command as you said. However, there are some considerations

* according to @chrmarti, for the `ubuntu` image, this file already existed with some content inside that he mentioned.
* for `alpine`, this file doesn't exist by default (from my observation so far). Thus, we need to create one with `auth sufficient ...` in order to make the `configureZshAsDefaultShell` work.

Not sure how we're gonna deal with this discrepancy though. Would appreciate your thoughts here.

Reply:
You could check what the behavior is when both lines are present and whether the order matters. Maybe the last entry wins or maybe duplicate entries result in an error.

Reply:
Hi @krikchaip, I believe this PR is still needed? 
Can you help address this comment, then we could help review.  Let us know if you need help, thanks!

Reply:
Yup. Sorry just got busy lately. I'm about to check what @chrmarti suggested. Will get back to you soon.

Reply:
@samruddhikhandale how about checking whether if `/etc/pam.d/chsh` exists? If that's the case then we look for a line with `auth (...) pam_shells.so` and replace `(...)` with `sufficient`. What do you guys think?

Reply:
Sounds good to me, and if `auth (...) pam_shells.so ` does not exist? then, append `auth sufficient pam_shells.so` to end.